module/os/freebsd/zfs/spa_os.c

```diff
@@ -94,12 +94,20 @@ spa_generate_rootconf(const char *name)
 	for (i = 0; i < count; i++) {
 		uint64_t txg;
 
+		if (configs[i] == NULL)
+			continue;
 		txg = fnvlist_lookup_uint64(configs[i], ZPOOL_CONFIG_POOL_TXG);
 		if (txg > best_txg) {
 			best_txg = txg;
 			best_cfg = configs[i];
 		}
 	}
+	if (best_txg == 0) {
```

Comment:
Thinking again, how can this even happen?  vdev_geom_read_pool_label() should return ENOENT if nothing is found.  There may be holes in the list. but there should be at least one element.

Reply:
I was thinking the element might be invalid, txg 0.

Reply:
OK. Does not sound very likely, but should not harm.

Reply:
Nevermind, configs with txg 0 are skipped elsewhere. I've removed this redundant check.